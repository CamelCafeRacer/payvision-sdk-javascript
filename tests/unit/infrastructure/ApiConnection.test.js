jest.mock('../../../src/infrastructure/HttpClient')
const HttpClient = require('../../../src/infrastructure/HttpClient')
const ApiConnection = require('../../../src/infrastructure/ApiConnection')

describe('ApiConnection', () => {
  let defaultConfig
  beforeEach(() => {
    defaultConfig = {
      username: 'username',
      password: 'password',
      businessId: 'businessId',
      baseUri: 'baseUri',
      debug: true,
    }
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('uses a default httpClient is none provided', async () => {
    const apiConnection = new ApiConnection(defaultConfig)
    expect(HttpClient).toHaveBeenCalledTimes(1)
    expect(apiConnection.httpClient instanceof HttpClient).toBe(true)
  })

  it('allows adding a custom httpClient', async () => {
    const httpClient = {
      config: jest.fn(),
      send: jest.fn(),
    }
    const apiConnection = new ApiConnection({
      ...defaultConfig,
      httpClient,
    })
    expect(HttpClient).toHaveBeenCalledTimes(0)
    expect(httpClient.send).toHaveBeenCalledTimes(0)
    expect(httpClient.config).toHaveBeenCalledTimes(1)
    expect(httpClient.config).toHaveBeenCalledWith({
      username: 'username',
      password: 'password',
      baseUri: 'baseUri',
      debug: true,
    })
    expect(apiConnection.httpClient instanceof HttpClient).toBe(false)
  })

  it('does basic validation of the httpClient in the constructor', () => {
    expect(() => new ApiConnection({
      ...defaultConfig,
      httpClient: {},
    })).toThrowErrorMatchingSnapshot()
    expect(() => new ApiConnection({
      ...defaultConfig,
      httpClient: {
        config: () => {},
      },
    })).toThrowErrorMatchingSnapshot()
  })
})
