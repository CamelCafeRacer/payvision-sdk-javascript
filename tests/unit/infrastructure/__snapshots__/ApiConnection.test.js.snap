// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiConnection does basic validation of the httpClient in the constructor 1`] = `"Custom httpClient needs to implement a config function"`;

exports[`ApiConnection does basic validation of the httpClient in the constructor 2`] = `"Custom httpClient needs to implement a send function"`;
